### PR TITLE
Feature/introduce stripe source

### PR DESCRIPTION
**Pull request by sanjushahgupta**
Following changes are made in this PR:
- added Stripe as an source option in factory.py.
- added StripeAnalyticsSource class in source.py to fetch data from stripe API
- added necessary dependencies to requirements.txt
- added ingestr.src.stripe_analytics as a module in pyproject.toml

### CODE REVIEW

File: ingestr/src/stripe_analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Stripe
```
Comment (karakanb) [inline ingestr/src/stripe_analytics/README.md:1]: we'd like to remove this README file from here.

File: ingestr/src/.dlt/.sources [hunk 2]
```diff
@@ -0,0 +1,28 @@
+engine_version: 1
```
Comment (karakanb) [inline ingestr/src/.dlt/.sources:1]: we'd like to remove the whole `.dlt` folder from here as well

Comment (sanjushahgupta) [reply ingestr/src/.dlt/.sources]: ok

File: ingestr/src/stripe_analytics/README.md (hunk 1 above)
Comment (sanjushahgupta) [reply ingestr/src/stripe_analytics/README.md]: ok

**Review by karakanb — APPROVED** [review]

**Pull request merged**